src/lib.rs

```diff
@@ -168,7 +171,87 @@ mod arch {
     }
 }
 
-#[cfg(not(target_os = "linux"))]
+#[cfg(target_os = "macos")]
+mod arch {
+    use super::*;
+
+    pub const SPAWN_ERROR: &str = "could not spawn xctrace";
+    pub const WAIT_ERROR: &str = "unable to wait for xctrace record child command to exit";
+
+    pub(crate) fn initial_command(
+        workload: Workload,
+        sudo: Option<Option<&str>>,
+        _freq: u32,
+        _custom_cmd: Option<String>,
+        verbose: bool,
+        ignore_status: bool,
+    ) -> Option<PathBuf> {
+        let xctrace = env::var("XCTRACE").unwrap_or_else(|_| "xctrace".to_string());
+        let trace_file = PathBuf::from("cargo-flamegraph.trace");
+        let mut command = sudo_command(&xctrace, sudo);
+        command
+            .arg("record")
+            .arg("--template")
+            .arg("Time Profiler")
+            .arg("--target-stdout")
+            .arg("-")
+            .arg("--output")
+            .arg(&trace_file);
+        match workload {
+            Workload::Command(args) => {
+                command.arg("--launch").arg("--").args(args);
+            }
+            Workload::Pid(pid) => {
+                match &*pid {
+                    [pid] => {
+                        // xctrace could accept multiple --attach <pid> arguments,
+                        // but it will only profiling on the last pid provided.
+                        command.arg("--attach").arg(pid.to_string());
+                    }
+                    _ => {
+                        panic!("xctrace only supports profiling a single process at a time");
```

Comment:
Can we `return Err(anyhow::anyhow!())` here like we do in other places?

Reply:
Originally it was `bail!()`. But `fn initial_command(` returns `Option`, which makes erroring impossible.

Reply:
I think wrapping those `Option`s in `Result` would make sense, yes. Please do so in a separate commit (either within this PR or in a separate PR).

Reply:
Resolved in #370